### PR TITLE
Only execute spinner function if there is a tty

### PR DESCRIPTION
**Pull request by rdwebdesign**
## Thank you for your contribution to the Pi-hole Community! 

Please read the comments below to help us consider your Pull Request.

We are all volunteers and completing the process outlined will help us review your commits quicker.

**Please make sure you**
 1. Base your code and PRs against the repositories developmental branch.
 2. [Sign Off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits as we enforce the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions

---
- **What does this PR aim to accomplish?:**

The `spinner` function was created to print only to `tty`, but it tried to print to `tty` even when there was nowhere to print.
In some cases, the error was causing the creation of a file as a side effect.
 
- **How does this PR accomplish the above?:**

Adding a condition to the `spinner` function.

- **What documentation changes (if any) are needed to support this PR?:**

none

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

### CODE REVIEW

File: advanced/Scripts/piholeDebug.sh [hunk 1]
```diff
@@ -1348,30 +1348,33 @@ check_database_integrity() {
...
-        fi
-        sleep 0.25
-    done
+    # Show the spinner only if there is a tty
+    if tty -s; then
+        local PID=$!  # PID of the most recent background process
+        local spin="/-\|"
+        local start=0
+        local elapsed=0
+        local i=1
```
Comment (dschaper) [inline advanced/Scripts/piholeDebug.sh:1357]: If you'd like to prepare for posix then you can use this suggestion. (There is no `local` in posix.)

```suggestion
    if tty -s; then
        _PID=$!  # PID of the most recent background process
        _spin="/-\|"
        _start=0
        _elapsed=0
        _i=1
```

File: advanced/Scripts/piholeDebug.sh [hunk 2]
```diff
@@ -1348,30 +1348,33 @@ check_database_integrity() {
...
+
+        start=$(date +%s) # Start the counter
```
Comment (dschaper) [inline advanced/Scripts/piholeDebug.sh:1359]: Just a style thing but can you move the inline comments to their own line?

File: advanced/Scripts/piholeDebug.sh (hunk 1 above)
Comment (rdwebdesign) [reply advanced/Scripts/piholeDebug.sh]: Done

File: advanced/Scripts/piholeDebug.sh (hunk 2 above)
Comment (rdwebdesign) [reply advanced/Scripts/piholeDebug.sh]: Done

**Review by dschaper — APPROVED** [review]

**Comment by yubiuser** [comment]
Needs rebase on `development` to remove Ubuntu 21 test.

**Comment by rdwebdesign** [comment]
> Needs rebase on `development` to remove Ubuntu 21 test.

I think my branch was updated. When did this happen?

**Comment by yubiuser** [comment]
It wasn't, that's way Ubnt21 failed.

**Review by yubiuser — APPROVED** [review]

**Pull request merged**